pkg/cmd/admin/migrate/storage/storage.go

```diff
@@ -0,0 +1,181 @@
+command to ensure that the most recent storage changes have been applied to all objects (storage
+version, storage encoding, any newer object defaults).
+
+To operate on a subset of resources, use the --include flag. If you encounter errors during a run
+the command will output a list of resources that received errors, which you can then re-run the
+command on. You may also specify --from-key and --to-key to restrict the set of resource names
+to operate on (key is NAMESPACE/NAME for resources in namespaces or NAME for cluster scoped
+resources). --from-key is inclusive if specified, while --to-key is exclusive.
+
+By default, events are not migrated since they expire within a very short period of time. If you
+have significantly increased the expiration time of events, run a migration with --include=events
+
+WARNING: This is a slow command and will put significant load on an API server. It may also
+  result in significant intra-cluster traffic.`
+
+	internalMigrateStorageExample = `  # Perform a dry-run of updating all objects
+  %[1]s
+
+  # To actually perform the update, the confirm flag must be appended
+  %[1]s --confirm
+
+  # Only migrate pods
+  %[1]s --include=pods --confirm
+
+  # Only pods that are in namespaces starting with "bar"
+  %[1]s --include=pods --confirm --from-key=bar/ --to-key=bar/\xFF
+`
+)
+
+type MigrateAPIStorageOptions struct {
+	migrate.ResourceOptions
+}
+
+// NewCmdMigrateAPIStorage implements a MigrateStorage command
+func NewCmdMigrateAPIStorage(name, fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &MigrateAPIStorageOptions{
+		ResourceOptions: migrate.ResourceOptions{
+			In:     in,
+			Out:    out,
+			ErrOut: errout,
+
+			Include: []string{"*"},
+			DefaultExcludes: []unversioned.GroupResource{
+				{Resource: "appliedclusterresourcequotas"},
+				{Resource: "bindings"},
+				{Resource: "deploymentconfigrollbacks"},
+				{Resource: "events"},
+				{Resource: "imagestreamimages"}, {Resource: "imagestreamtags"}, {Resource: "imagestreammappings"}, {Resource: "imagestreamimports"},
+				{Resource: "projectrequests"}, {Resource: "projects"},
+				{Resource: "componentstatuses"},
+				{Resource: "clusterrolebindings"}, {Resource: "rolebindings"},
+				{Resource: "clusterroles"}, {Resource: "roles"},
+				{Resource: "resourceaccessreviews"}, {Resource: "localresourceaccessreviews"}, {Resource: "subjectaccessreviews"},
+				{Resource: "selfsubjectrulesreviews"}, {Resource: "localsubjectaccessreviews"},
+				{Resource: "replicationcontrollerdummies.extensions"},
+				{Resource: "podtemplates"},
+				{Resource: "useridentitymappings"},
+			},
+			// Resources known to share the same storage
+			OverlappingResources: []sets.String{
+				sets.NewString("horizontalpodautoscalers.autoscaling", "horizontalpodautoscalers.extensions"),
+				sets.NewString("jobs.batch", "jobs.extensions"),
```

Comment:
@deads2k @mfojtik if this merges before your D/DC stuff it should be updated as well.